crates/manifest/src/schema/common.rs

```diff
@@ -30,13 +30,33 @@ pub enum ComponentSource {
         /// `digest = `"sha256:abc123..."`
         digest: String,
     },
+    /// `{ ... }`
+    Registry {
+        /// `registry = "example.com"`
+        registry: Option<String>,
+        /// `package = "example:component"`
+        package: String,
+        /// `version = "1.2.3"`
+        version: semver::Version,
+    },
 }
 
 impl Display for ComponentSource {
     fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
         match self {
             ComponentSource::Local(path) => write!(f, "{path:?}"),
             ComponentSource::Remote { url, digest } => write!(f, "{url:?} with digest {digest:?}"),
+            ComponentSource::Registry {
+                registry,
+                package,
+                version,
+            } => {
+                let registry_suffix = match registry {
+                    None => "default registry".to_owned(),
+                    Some(r) => format!("registry {r:?}"),
+                };
+                write!(f, "\"{package}@{version}\" from {registry_suffix:?}")
```

Comment:
```suggestion
                write!(f, "\"{package}@{version}\" from {registry_suffix}")
```

Reply:
I PUT THAT FORMATTING IN SPECIALLY FOR YOU LANN

Reply:
DOUBLE THE QUOTES, DOUBLE THE ENLIGHTENMENT